.github/workflows/tics.yml

```
@@ -0,0 +1,72 @@
+name: Linux
+
+on:
+    schedule:
+    # 0:00 UTC every Sunday
+    - cron: 0 0 * * 0
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

You might want to opt for our hosted runners. Should be quicker and more reliable.

```suggestion
    runs-on: [self-hosted, reactive, amd64, large-tiobe, noble]
```

Some other tweaks I needed / added over past weeks:

https://github.com/canonical/mir/compare/0a737ee2c1a1186266391dd8d26096e492587e0f...main#diff-6f8e29420dabe4694a23e1b4d06801de1b54ea481b379216a51d41213296bdf5

---

Hey @Saviq, thanks for the heads-up! The diff concerns mainly running in PRs, right? Is there anything else we should note?

@sharder996, did you mean to change the runner in a separate PR?

---

I can change the runner here and any other small tweaks you see.

---

Yes, the most relevant here would be `runs-on` (I actually [tried `2xlarge` since](https://github.com/canonical/mir/actions/runs/14515905858) and it halved the analysis time).

The `--yes`es are required on our self hosted runners - GH probably has tweaked APT config.